### PR TITLE
[Feature] Support iSAID dataset

### PR DESCRIPTION
**Pull request by Renzhihan**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support iSAID instance segmentation dataset for mmdet. 
iSAID link:https://captain-whu.github.io/iSAID/dataset.html .
Some code references the implementation in mmseg.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Renzhihan** [comment]
The main features has been completed and I will add an English README file as soon as possible

**Comment by hhaAndroid** [comment]
@Renzhihan please fix the conflicts

File: mmdet/datasets/isaid.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class iSAIDDataset(CocoDataset):
+    """Dataset for iSAID instance segmentation."""
+
+    METAINFO = dict(
+        classes=('background', 'ship', 'store_tank', 'baseball_diamond',
+                 'tennis_court', 'basketball_court', 'Ground_Track_Field',
+                 'Bridge', 'Large_Vehicle', 'Small_Vehicle', 'Helicopter',
+                 'Swimming_pool', 'Roundabout', 'Soccer_ball_field', 'plane',
+                 'Harbor'),
+        palette=[[0, 0, 0], [0, 0, 63], [0, 63, 63], [0, 63, 0], [0, 63, 127],
```
Comment (hhaAndroid) [inline mmdet/datasets/isaid.py:16]:
```suggestion
        palette=[(0, 0, 0), (0, 0, 63), ...,
```

File: mmdet/datasets/isaid.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmdet.registry import DATASETS
+from .coco import CocoDataset
+
+
+@DATASETS.register_module()
+class iSAIDDataset(CocoDataset):
+    """Dataset for iSAID instance segmentation."""
```
Comment (hhaAndroid) [inline mmdet/datasets/isaid.py:8]: Please write a link to the dataset and related information

File: mmdet/datasets/isaid.py (hunk 1 above)
Comment (hhaAndroid) [reply mmdet/datasets/isaid.py:16]: list[list] -> list[tuple]

File: projects/iSAID/configs/mask_rcnn_r50_fpn_1x_isaid.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+_base_ = [
```
Comment (hhaAndroid) [inline projects/iSAID/configs/mask_rcnn_r50_fpn_1x_isaid.py:1]: Do you have some training results as a reference?

**Comment by hhaAndroid** [comment]
@Renzhihan Please fix the conflicts.

**Pull request merged**